internal/ethapi/simulate.go

```diff
@@ -359,12 +361,37 @@ func (sim *simulator) processBlock(ctx context.Context, block *simBlock, header,
 		reqHash := types.CalcRequestsHash(requests)
 		header.RequestsHash = &reqHash
 	}
-	blockBody := &types.Body{Transactions: txes, Withdrawals: *block.BlockOverrides.Withdrawals}
+
+	// For OP Stack Jovian blocks, inject a synthetic deposit transaction at the beginning of the block.
+	// This is required because CalcDAFootprint (called by FinalizeAndAssemble for Jovian blocks)
+	// expects the first transaction to be a deposit transaction containing L1 attributes data.
+	isJovian := sim.chainConfig.IsJovian(header.Time)
+	finalTxes := txes
+	if isJovian {
+		depositTx := createSimulatedJovianDepositTx()
+		finalTxes = append([]*types.Transaction{depositTx}, txes...)
+	}
```

Comment:
If you look at this test code: 
https://github.com/ethereum-optimism/op-geth/blob/472228745b7e5c6d777897c83067fe9dacc47ad3/miner/miner_optimism_test.go#L116-L118

you can see we don't insert a jovian deposit tx until the block after the activation block.

Also, you might be able to move the helper function from the test so that it can be used here as well?